api/firewaller/machine.go

```diff
@@ -163,3 +168,49 @@ func (m *Machine) IsManual() (bool, error) {
 	}
 	return result.Result, nil
 }
+
+// OpenedMachinePortRanges queries the open port ranges for all units on this
+// machine and returns back a map where keys are unit names and values are maps
+// of the opened port ranges by the unit grouped by subnet CIDR.
+func (m *Machine) OpenedMachinePortRanges() (map[names.UnitTag]map[string][]network.PortRange, error) {
```

Comment:
`map[string][]network.PortRange` appears quite a lot. Might we make a type/alias for it?

Reply:
Thought of it but was unsure how to call it. However, note that just as with the previous PR, the idea is to feed this into a helper type that exposes a friendly API so you are not going to be seeing much of it floating around in the code-base.

Reply:
After thinking about it, it might make sense to add this to core network as a `GroupedPortRanges` type. Then I can provide the UniquePortRanges helper for it and clean up the repeated code from the call-sites.

Reply:
:+1: